paddle/operators/softmax_op.cc

```diff
@@ -40,7 +40,17 @@ class SoftmaxOpMaker : public OpProtoAndCheckerMaker {
 
 class SoftmaxOpGrad : public OperatorWithKernel {
 protected:
-  void InferShape(const InferShapeContext &ctx) const override {}
+  void InferShape(const InferShapeContext &ctx) const override {
+    PADDLE_ENFORCE(ctx.InputSize() == 2, "Input of SoftmaxOpGrad should be 2");
+    PADDLE_ENFORCE(ctx.OutputSize() == 1,
+                   "Output of SoftmaxOpGrad should be 1");
+    PADDLE_ENFORCE(ctx.InputVar(0) != nullptr, "Input(0) should not be null");
+    PADDLE_ENFORCE(ctx.InputVar(1) != nullptr, "Input(1) should not be null");
+    PADDLE_ENFORCE(ctx.Input<Tensor>(0)->dims() == ctx.Input<Tensor>(1)->dims(),
+                   "the shape of Input(0) and Input(1) should be the same");
+    ctx.Output<Tensor>(0)->ResizeLike(*ctx.Input<Tensor>(0));
```

Comment:
I do not think `ResizeLike` method is needed. Because it does not simply our implementation.